ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -1008,52 +994,62 @@ std::string McapHandler::serialize_qos_(
 }
 
 std::string McapHandler::serialize_type_identifier_(
-        const eprosima::fastrtps::types::TypeIdentifier* type_identifier)
+        const fastdds::dds::xtypes::TypeIdentifier& type_identifier)
 {
     // Reserve payload and create buffer
-    size_t size = fastrtps::types::TypeIdentifier::getCdrSerializedSize(*type_identifier) +
-            eprosima::fastrtps::rtps::SerializedPayload_t::representation_header_size;
-    fastrtps::rtps::SerializedPayload_t payload(static_cast<uint32_t>(size));
-    eprosima::fastcdr::FastBuffer fastbuffer((char*) payload.data, payload.max_size);
+    fastcdr::CdrSizeCalculator calculator(fastcdr::CdrVersion::XCDRv2);
+    size_t current_alignment {0};
+    size_t size = calculator.calculate_serialized_size(type_identifier, current_alignment) +
+                            fastdds::rtps::SerializedPayload_t::representation_header_size;
+
+    fastdds::rtps::SerializedPayload_t payload(static_cast<uint32_t>(size));
+    fastcdr::FastBuffer fastbuffer((char*) payload.data, payload.max_size);
 
     // Create CDR serializer
-    #if FASTRTPS_VERSION_MAJOR <= 2 && FASTRTPS_VERSION_MINOR < 13
-    eprosima::fastcdr::Cdr ser(fastbuffer, eprosima::fastcdr::Cdr::DEFAULT_ENDIAN, eprosima::fastcdr::Cdr::DDS_CDR);
-    #else
-    eprosima::fastcdr::Cdr ser(fastbuffer, eprosima::fastcdr::Cdr::DEFAULT_ENDIAN,
-            eprosima::fastcdr::CdrVersion::XCDRv1);
-    #endif // if FASTRTPS_VERSION_MAJOR <= 2 && FASTRTPS_VERSION_MINOR < 13
+    fastcdr::Cdr ser(fastbuffer, fastcdr::Cdr::DEFAULT_ENDIAN,
+            fastcdr::CdrVersion::XCDRv2);
 
-    payload.encapsulation = ser.endianness() == eprosima::fastcdr::Cdr::BIG_ENDIANNESS ? CDR_BE : CDR_LE;
+    payload.encapsulation = ser.endianness() == fastcdr::Cdr::BIG_ENDIANNESS ? CDR_BE : CDR_LE;
 
     // Serialize
-    type_identifier->serialize(ser);
-#if FASTCDR_VERSION_MAJOR == 1
-    payload.length = (uint32_t)ser.getSerializedDataLength();
-    size = (ser.getSerializedDataLength() + 3) & ~3;
-#else
+    fastcdr::serialize(ser, type_identifier);
     payload.length = (uint32_t)ser.get_serialized_data_length();
     size = (ser.get_serialized_data_length() + 3) & ~3;
-#endif // if FASTCDR_VERSION_MAJOR == 1
-
-    // Create CDR message
-    // NOTE: Use 0 length to avoid allocation (memory already reserved in payload creation)
-    eprosima::fastrtps::rtps::CDRMessage_t* cdr_message = new eprosima::fastrtps::rtps::CDRMessage_t(0);
-    cdr_message->buffer = payload.data;
-    cdr_message->max_size = payload.max_size;
-    cdr_message->length = payload.length;
-#if __BIG_ENDIAN__
-    cdr_message->msg_endian = eprosima::fastrtps::rtps::BIGEND;
-#else
-    cdr_message->msg_endian = eprosima::fastrtps::rtps::LITTLEEND;
-#endif // if __BIG_ENDIAN__
+
+    // Create CDR message with payload
+    fastdds::rtps::CDRMessage_t* cdr_message = new fastdds::rtps::CDRMessage_t(payload);
```

Comment:
NIT: you can remove the wraps thing at the end since this constructor already sets it.